src/VideoReceiver/GStreamer.cc

```diff
@@ -105,20 +105,58 @@ static void qgcputenv(const QString& key, const QString& root, const QString& pa
 }
 #endif
 
-static void
-blacklist()
+void
+GStreamer::blacklist(bool forceVAAPI, bool forceNVIDIA, bool forceD3D11)
 {
-    GstRegistry* reg;
+    GstRegistry* registry = gst_registry_get();
+
+    if (registry == nullptr) {
+        qCCritical(GStreamerLog) << "Failed to get gstreamer registry.";
+        return;
+    }
+
+    auto changeRank = [registry](const char* featureName, uint16_t rank) {
+        GstPluginFeature* feature = gst_registry_lookup_feature(registry, featureName);
+        if (feature == nullptr) {
+            qCDebug(GStreamerLog) << "Failed to change ranking of feature:" << featureName;
+            return;
+        }
+
+        qCInfo(GStreamerLog) << "Changing feature (" << featureName << ") to use rank:" << rank;
+        gst_plugin_feature_set_rank(feature, rank);
+        gst_registry_add_feature(registry, feature);
+        gst_object_unref(feature);
+    };
 
-    if ((reg = gst_registry_get()) == nullptr) {
+    // Set rank for specific features
+    changeRank("bcmdec", GST_RANK_NONE);
+
+    if (!forceVAAPI && !forceNVIDIA && !forceD3D11) {
         return;
     }
 
-    GstPluginFeature* plugin;
+    // Decrease software rank to not interfere with next plugins
+    changeRank("avdec_h264", GST_RANK_MARGINAL + 1);
+
+    // Enable VAAPI drivers
+    if (forceVAAPI) {
+        for(auto name : {"vaapimpeg2dec", "vaapimpeg4dec", "vaapih263dec", "vaapih264dec", "vaapivc1dec", "vaapidecodebin"}) {
+            changeRank(name, GST_RANK_PRIMARY);
```

Comment:
If what if there is another h264 decoder that has primary rank? We aren't forcing here in that case. See my other suggestion in the lines above.

Reply:
That's all vaapi decoders.